js/shared_controls.js

```diff
@@ -571,9 +571,29 @@ function Pokemon(pokeInfo) {
 		];
 		this.weight = +pokeInfo.find(".weight").val();
 	}
-	this.hasType = function(type) {
+	this.hasAbility = function (arguments) {
```

Comment:
@_@ You don't need to declare ``arguments``. It's automatically defined.

Reply:
I know. This was a year ago, I was only passing it to see if it would work because I couldn’t get anywhere and I was grasping a straws

Reply:
Probably should’ve reverted it back to what it was originally before I “gave up” since that code wasn’t garbage but yeah

Reply:
this should be better, I have no idea why I had so many issues with this, I think I just got flustered.